src/IO/H5/VolumeData.cpp

```diff
@@ -306,6 +306,29 @@ std::vector<std::vector<size_t>> VolumeData::get_extents(
   return individual_extents;
 }
 
+std::pair<size_t, size_t> offset_and_length_for_grid(
+    const std::string& grid_name,
+    const std::vector<std::string>& all_grid_names,
+    const std::vector<std::vector<size_t>>& all_extents) noexcept {
+  auto found_grid_name =
+      std::find(all_grid_names.begin(), all_grid_names.end(), grid_name);
+  if (found_grid_name == all_grid_names.end()) {
+    ERROR("Found no grid named '" + grid_name + "'.");
+  } else {
+    const auto element_index =
+        std::distance(all_grid_names.begin(), found_grid_name);
+    size_t element_data_offset = std::accumulate(
+        all_extents.begin(), all_extents.begin() + element_index,
+        size_t{0}, [](size_t offset, std::vector<size_t> extents) noexcept {
```

Comment:
we should use `static_cast<size_t>(0)` in these cases to avoid implicit conversion, or (easier to read), include `Utilities/Literals.hpp` which allows you to do `0_st`